lib/poller.py

```diff
@@ -0,0 +1,53 @@
+from lib.redis_client import RedisClient
+from lib.sierra_db_client import SierraDbClient
+from nypl_py_utils.functions.log_helper import create_log
+
+
+class Poller:
+    """
+    Poll Serra db for holdshelf events and trigger notifications to patrons.
+    """
+
+    def __init__(self):
+        self.logger = create_log('poller')
+
+        self.sierra_client = SierraDbClient()
+        self.redis_client = RedisClient()
+
+    def poll(self):
+        """Retrieve holdshelf entries from Sierra, remove processed entries,
+        and send notifications for what remains.
+        """
+        self.logger.info('Starting polling')
+
+        entries = self.sierra_client.holdshelf_entries()
+
+        self.logger.info(f'Found {len(entries)} holdshelf entries')
+        try:
+            entries = [entry for entry in entries if self.unprocessed(entry)]
```

Comment:
(Optional) You can use the built-in filter() method instead of doing list comprehension